app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -0,0 +1,136 @@
+;(function (Modules, root) {
+  'use strict'
+
+  var $ = root.$
+  var $window = $(root)
+
+  Modules.HighlightActiveSectionHeading = function () {
+    var self = this
+    var _hasResized = true
+    var _hasScrolled = true
+    var _interval = 50
+    var anchorIDs = []
+
+    self.getWindowDimensions = function () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      }
+    }
+
+    self.getWindowPositions = function () {
+      return {
+        scrollTop: $window.scrollTop()
+      }
+    }
+
+    self.getElementOffset = function ($el) {
+      return $el.offset()
+    }
+
+    self.start = function ($el) {
+      $window.resize(self.hasResized)
+      $window.scroll(self.hasScrolled)
+
+      setInterval(self.checkResize, _interval)
+      setInterval(self.checkScroll, _interval)
+
+      self.$anchors = $el.find('.js-page-contents a')
+      self.totalAnchors = self.$anchors.length
+      self.getAnchors(self.totalAnchors)
```

Comment:
getAnchors is returning anchorIDs, but the result just gets thrown away? I'd suggest either removing the `return` from `getAnchors` to make it clear it's manipulating a variable outside of its own scope, or assigning the result to anchorIDs here.